backend/hitas/views/housing_company.py

```diff
@@ -52,33 +51,79 @@ class HousingCompanyDetailSerializer(EnumSupportSerializerMixin, serializers.Mod
     area = serializers.SerializerMethodField()
     date = serializers.SerializerMethodField()
     real_estates = serializers.SerializerMethodField()
+    building_type = serializers.SerializerMethodField()
+    financing_method = serializers.SerializerMethodField()
+    developer = serializers.SerializerMethodField()
+    property_manager = serializers.SerializerMethodField()
+    acquisition_price = serializers.SerializerMethodField()
+    primary_loan = serializers.SerializerMethodField()
+    last_modified = serializers.SerializerMethodField()
+    notes = serializers.SerializerMethodField()
+    legacy_id = serializers.SerializerMethodField()
+
+    def get_id(self, obj: HousingCompany) -> str:
+        return obj.uuid.hex
 
-    def get_name(self, obj):
+    def get_name(self, obj: HousingCompany) -> Dict[str, str]:
         return {
             "display": obj.display_name,
             "official": obj.official_name,
         }
 
-    def get_id(self, obj):
-        return obj.uuid.hex
+    def get_legacy_id(self, obj: HousingCompany) -> Optional[str]:
+        return value_or_null(obj.legacy_id)
+
+    def get_notes(self, obj: HousingCompany) -> Optional[str]:
+        return value_or_null(obj.notes)
 
-    def get_state(self, obj):
+    def get_state(self, obj: HousingCompany) -> str:
         return obj.state.name.lower()
 
-    def get_address(self, obj):
-        return {
-            "street": obj.street_address,
-            "postal_code": obj.postal_code.value,
-            "city": obj.city,
-        }
+    def get_address(self, obj: HousingCompany) -> Address:
+        return address_obj(obj)
 
-    def get_area(self, obj):
+    def get_area(self, obj: HousingCompany) -> Dict[str, any]:
         return {"name": obj.postal_code.description, "cost_area": obj.area}
 
-    def get_date(self, obj):
+    def get_date(self, obj: HousingCompany) -> datetime.date:
         return obj.date
 
-    def get_real_estates(self, obj):
+    def get_building_type(self, obj: HousingCompany) -> Code:
+        return code_to_obj(obj.building_type)
+
+    def get_financing_method(self, obj: HousingCompany) -> Code:
+        return code_to_obj(obj.financing_method)
+
+    def get_developer(self, obj: HousingCompany) -> Code:
+        return code_to_obj(obj.developer)
```

Comment:
I would create serializers for all related objects being turned into dicts instead